third_party/pcsc-lite/naclport/server_clients_management/src/admin-policy-service.js

```diff
@@ -0,0 +1,207 @@
+
+/**
+ * This class provides the C++ layer with the AdminPolicy obtained from the
+ * managed storage. Upon initialization, it reads the current value from
+ * storage and sends a message containing the policy data.
+ * @param {!goog.messaging.AbstractChannel} serverMessageChannel Message channel
+ * to the PC/SC server into which the admin policy will be forwarded.
+ * @param {!ReadinessTracker} serverReadinessTracker Tracker of the PC/SC server
+ * that allows to delay forwarding of the messages to the server until it is
+ * ready to receive them.
+ * @constructor
+ * @extends goog.Disposable
+ */
+GSC.PcscLiteServerClientsManagement.AdminPolicyService = function(
+    serverMessageChannel, serverReadinessTracker) {
+  AdminPolicyService.base(this, 'constructor');
+
+  /**
+   * @type {!goog.log.Logger}
+   * @const
+   */
+  this.logger = GSC.Logging.getScopedLogger(
+      'PcscLiteServerClientsManagement.AdminPolicyService');
+
+  /** @private */
+  this.serverMessageChannel_ = serverMessageChannel;
+
+  /** @private */
+  this.serverReadinessTracker_ = serverReadinessTracker;
+
+  // The policy update messages are deferred until the server readiness is
+  // established.
+  /** @private */
+  this.deferredProcessor_ =
+      new DeferredProcessor(this.serverReadinessTracker_.promise);
+
+  this.addChannelDisposedListener_();
+
+  /**
+   * @type {!Object}
+   * @private
+   */
+  this.adminPolicy_ = {};
+
+  this.loadManagedStorage_();
+  this.listenForStorageChanging_();
+
+  goog.log.fine(this.logger, 'Initialized');
+};
+
+const AdminPolicyService =
+    GSC.PcscLiteServerClientsManagement.AdminPolicyService;
+
+goog.inherits(AdminPolicyService, goog.Disposable);
+
+/** @override */
+AdminPolicyService.prototype.disposeInternal = function() {
+  this.deferredProcessor_.dispose();
+  this.deferredProcessor_ = null;
+
+  this.serverReadinessTracker_ = null;
+
+  this.serverMessageChannel_ = null;
+
+  goog.log.fine(this.logger, 'Disposed');
+
+  AdminPolicyService.base(this, 'disposeInternal');
+};
+
+/** @private */
+AdminPolicyService.prototype.loadManagedStorage_ = function() {
+  chrome.storage.managed.get(this.managedStorageLoadedCallback_.bind(this));
+};
+
+/**
+ * @param {!Object} items
+ * @private
+ */
+AdminPolicyService.prototype.managedStorageLoadedCallback_ = function(items) {
+  goog.log.fine(
+      this.logger,
+      'Loaded the following data from the managed storage: ' +
+          GSC.DebugDump.dump(items));
+
+  this.updateAdminPolicyFromStorageData_(items);
+};
+
+/**
+ * Set up listener tracking storage changes.
+ * @private
+ */
+AdminPolicyService.prototype.listenForStorageChanging_ = function() {
+  chrome.storage.managed.onChanged.addListener(
+      this.storageChangedListener_.bind(this));
+};
+
+/**
+ * @param {!Object} changes
+ * @private
+ */
+AdminPolicyService.prototype.storageChangedListener_ = function(changes) {
+  if (this.isDisposed()) return;
```

Comment:
nit: Is this the style that clang-format uses? If not, then please reformat the code by running "./format-code.sh".